sdk/trace/batch_span_processor_test.go

```diff
@@ -209,7 +209,8 @@ func generateSpan(t *testing.T, parallel bool, tr trace.Tracer, option testOptio
 
 	wg := &sync.WaitGroup{}
 	for i := 0; i < option.genNumSpans; i++ {
-		binary.BigEndian.PutUint64(sc.TraceID[0:8], uint64(i+1))
+		tid := sc.TraceID()
+		binary.BigEndian.PutUint64(tid[0:8], uint64(i+1))
```

Comment:
Should `sc` be assigned a new SpanContext with this `tid`?

E.g.

```suggestion
		binary.BigEndian.PutUint64(tid[0:8], uint64(i+1))
		newSC = sc.WithTraceID(tid)
```

And that SpanContext would be passed to `trace.ContextWithRemoteSpanContext`?